src/CMakeLists.txt

```diff
@@ -104,6 +104,8 @@ if(NOT ENABLE_SYSTEM_LUA)
 		# silence a Clang specific warning when compiling the lua code
 		set_property(SOURCE SOURCE ${lua_sources} APPEND_STRING PROPERTY COMPILE_FLAGS " -Wno-string-plus-int")
 	endif()
+
+	set_source_files_properties(${lua_sources} PROPERTIES SKIP_LINTING ON)
```

Comment:
If this just skips `src/modules/lua` then it should skip `src/modules` instead. Any code under `src/modules` is not from us.

Reply:
Strictly speaking, this doesn't skip src/modules/lua, it sets the SKIP_LINTING attribute on all of the source files from lua that we're compiling.

I refactored the CMakeLists.txt to allow for enabling/disabling linting based on the directory.